oshmem/mca/atomic/atomic.h

```diff
@@ -86,17 +86,26 @@ struct mca_atomic_base_module_1_0_0_t {
     opal_object_t super;
 
     /* Collective function pointers */
+    int (*atomic_add)(void *target,
+                      uint64_t value,
+                      size_t size,
+                      int pe);
     int (*atomic_fadd)(void *target,
                        void *prev,
-                       const void *value,
-                       size_t nlong,
-                       int pe,
-                       struct oshmem_op_t *op);
+                       uint64_t value,
+                       size_t size,
+                       int pe);
+    int (*atomic_swap)(void *target,
+                       void *prev,
+                       uint64_t value,
+                       size_t size,
+                       int pe);
     int (*atomic_cswap)(void *target,
-                        void *prev,
-                        const void *cond,
-                        const void *value,
-                        size_t nlong,
+                        uint64_t *prev, /* prev is used internally by wrapper, we may
```

Comment:
let's do the same for other atomics (fadd, swap)

Reply:
why?
in other calls there is no postprocessing of prev value - it is updated directly from ucx_atomic calls

Reply:
i see.. but in latest patch there is also no postprocessing for cswap.. or did i miss it? maybe it's not needed at all

Reply:
in force of result value to uint64_t postprocessing of result value is not required anymore - we know that there is no memory corruption :)